tests/test_updater_with_simulator.py

```diff
@@ -257,6 +257,44 @@ def test_not_loading_targets_twice(self, wrapped_open: MagicMock) -> None:
         updater.get_targetinfo("somepath")
         wrapped_open.assert_not_called()
 
+    @patch.object(builtins, "open", wraps=builtins.open)
+    def test_update_metadata_to_cache(self, wrapped_open: MagicMock) -> None:
+        updater = self._run_refresh()
+
+        # Add targets to repository
+        self.sim.targets.version += 1
+        self.sim.add_target("targets", b'some content', self.targets_dir)
+        self.sim.update_snapshot()
+
+        # Make a successful update of valid metadata which stores it in cache
+        updater = self._run_refresh()
+
+        # Clean up calls to fetch during refresh()
+        wrapped_open.reset_mock()
+
+        updater.get_targetinfo(self.targets_dir)
+
+        # Create a new updater and perform a second update while
+        # the metadata is already stored in cache (metadata dir)
+        updater = Updater(
+            self.metadata_dir,
+            "https://example.com/metadata/",
+            self.targets_dir,
+            "https://example.com/targets/",
+            self.sim,
+        )
+        updater.get_targetinfo(self.targets_dir)
```

Comment:
Calling `get_targetinfo()` feels correct but the argument is confusing: maybe just use `"non existent target"` or something like that as target_path -- the value does not matter but using `targets_dir` is misleading.